File: _old/15_Concatenation_Value/Golang/Main.go
package main

import (
	"fmt"
	"strconv"
)

func findTheArrayConcVal(nums []int) int64 {
	var result int64 = 0

	for i := 0; i < len(nums); i++ {
		if len(nums) > 1 {
			value, _ := strconv.Atoi(strconv.Itoa(nums[0]) + strconv.Itoa(nums[len(nums)-1]))
			result += int64(value)
			nums = nums[1:]
			nums = nums[:len(nums)-1]
			i--
		} else {
			fmt.Println(nums[i])
			result += int64(nums[i])
		}
	}

	return result
}

func main() {

	x := findTheArrayConcVal([]int{5, 14, 13, 8, 12})
	fmt.Println(x)
}

File: _old/15_Concatenation_Value/JavaScript/Main.js
let findConcatenationValue = function (nums) {           
  let result = 0

  for (let i = 0; i < nums.length; i++) {
    if (nums.length > 1) {
      let value = String(nums[0]) + String(nums[nums.length - 1])
      result += Number(value)
      nums.shift()
      nums.pop()
      i--
    } else {
      result += nums[i]
    }
  }

  return result
}

console.log(findConcatenationValue([5, 14, 13, 8, 12]));
